File: AtheLoader/test/wrapper/wrapper.cpp
#include "wrapper.h"
#include "stdio.h"
#include "string.h"
#include "stdlib.h"

/*
 * 用于实现wrapper demo的mock版本
 * 目前仅支持同步插件接口,非实时结果返回模式的mock
 * */

const void* wrapperInnerHdl = "wrapperTestHandle";
pDataList wrapperInnerRslt = NULL;
pDataList wrapperOnceRslt = NULL;

int WrapperAPI wrapperInit(pConfig cfg){
	printf("wrapperInit\n");
    
    // 打印输入配置项
    while (cfg != NULL) {
        printf("key=%s, value=%s\n", cfg->key, cfg->value);
        cfg = cfg->next;
    }

    // 构建内部测试read值
    wrapperInnerRslt = (struct DataList*)malloc(sizeof(struct DataList));
    wrapperInnerRslt->key = (char*)"result";
    wrapperInnerRslt->data = (void*)"response result from AtheLoader wrapper";
    wrapperInnerRslt->len = strlen("response result from AtheLoader wrapper");
    wrapperInnerRslt->desc = NULL;
    wrapperInnerRslt->encoding = (char*)"utf8";
    wrapperInnerRslt->type = DataText;
    wrapperInnerRslt->status = DataEnd;
    wrapperInnerRslt->next = NULL;

    wrapperOnceRslt = (struct DataList*)malloc(sizeof(struct DataList));
    wrapperOnceRslt->key = (char*)"result";
    wrapperOnceRslt->data = (void*)"response result from AtheLoader wrapper";
    wrapperOnceRslt->len = strlen("response result from AtheLoader wrapper");
    wrapperOnceRslt->desc = NULL;
    wrapperOnceRslt->encoding = (char*)"utf8";
    wrapperOnceRslt->type = DataText;
    wrapperOnceRslt->status = DataOnce;
    wrapperOnceRslt->next = NULL;
    return 0;
}

int WrapperAPI wrapperFini(){
	printf("wrapperFini\n");
	return 0;
}

const char* WrapperAPI wrapperError(int errNum){
    printf("wrapperError\n");	
    return "inner error";
}

const char* WrapperAPI wrapperVersion(){
	printf("wrapperVersion\n");
    return "1.0.0";
}

int WrapperAPI wrapperLoadRes(pDataList perData, unsigned int resId){
	printf("wrapperLoadRes\n");
	return 0;
}

int WrapperAPI wrapperUnloadRes(unsigned int resId){
	printf("wrapperUnloadRes\n");
	return 0;
}

const void* WrapperAPI wrapperCreate(const void* usrTag, pParamList params, wrapperCallback cb, unsigned int psrIds[], int psrCnt, int* errNum){
	printf("wrapperCreate\n");
	return wrapperInnerHdl;
}

int WrapperAPI wrapperWrite(const void* handle, pDataList reqData){
	printf("wrapperWrite\n");
	return 0;
}

int WrapperAPI wrapperRead(const void* handle, pDataList* respData){
	printf("wrapperRead\n");
    /*
Note: demo 关闭实时读功能,wrapperRead仅需返回最终last结果,
    若需实时返回结果,可通过修改配置[aiges].realTimeRlt = true
     */
    respData = &wrapperInnerRslt; 
	return 0;
}

int WrapperAPI wrapperDestroy(const void* handle){
	printf("wrapperDestroy\n");
	return 0;
}

int WrapperAPI wrapperExec(pParamList params, pDataList reqData, pDataList* respData){
	printf("wrapperExec\n");
	respData = &wrapperOnceRslt;
    return 0;
}

int WrapperAPI wrapperExecFree(pDataList* respData){
	printf("wrapperExecFree\n");
	return 0;
}

int WrapperAPI wrapperExecAsync(const void* usrTag, pParamList params, pDataList reqData, wrapperCallback callback, int timeout){
	printf("wrapperExecAsync\n");
	return 0;
}

const char* WrapperAPI wrapperDebugInfo(const void* handle){
	printf("wrapperDebugInfo\n");
	return "DebugInfo";
}

File: demo/wrapper/wrapper.py
import sys
if not hasattr(sys, 'argv'):
    sys.argv  = ['']


'''
服务初始化
@param config:
    插件初始化需要的一些配置，字典类型
    key: 配置名
    value: 配置的值
@return
    ret: 错误码。无错误时返回0
'''
def wrapperInit(config: {}) -> int:
    return 0


'''
服务逆初始化

@return
    ret:错误码。无错误码时返回0
'''
def wrapperFini() -> int:
    return 0

'''
非会话模式计算接口,对应oneShot请求,可能存在并发调用

@param usrTag 句柄
#param params 功能参数
@param  reqData     写入数据实体
@param  respData    返回结果实体,内存由底层服务层申请维护,通过execFree()接口释放
@param psrIds 需要使用的个性化资源标识列表
@param psrCnt 需要使用的个性化资源个数

@return 接口错误码
    reqDat
    ret:错误码。无错误码时返回0
'''
def wrapperOnceExec(usrTag:str,params:{},reqData:[],respData:[],psrIds:[],psrCnt:int) -> int:
    print("hello world")
    print(usrTag)
    print(params)
    print(reqData)
    print(psrIds)
    print(psrCnt)
    return 100


def wrapperCreate(usrTag: str, params: [], psrIds: [], psrCnt: int) -> str:
    return ""


def wrapperWrite(handle: str, datas: []) -> int:
    return 0


def wrapperRead(handle: str) -> []:
    return []


def wrapperDestroy(handle: str) -> int:
    return 0


def wrapperError(ret:int)->str:
    if ret==100:
        return "this is a tese error return"
    return ""